zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ElasticsearchBatchRequest.java

```diff
@@ -230,6 +236,11 @@ public void executeWithRefresh() throws PersistenceException {
     execute(true);
   }
 
+  @Override
+  public void onError(final String index, final Consumer<String> errorHandler) {
+    errorHandlers.put(index, errorHandler);
+  }
+
   private void execute(final boolean shouldRefresh) throws PersistenceException {
```

Comment:
If you ask me. I would accept here an error handler, and call it in the execute method if an error appears.

Instead of the indirection, where the RecordHandler adds themself as error handlers, then calling all of them here. I would do some more logic in the batch writer.